presto-jdbc/src/test/java/io/prestosql/jdbc/TestJdbcResultSetTimezone.java

```diff
@@ -0,0 +1,384 @@
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.Date;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.sql.Time;
+import java.sql.Timestamp;
+import java.sql.Types;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.ZoneId;
+import java.util.Calendar;
+import java.util.List;
+import java.util.TimeZone;
+
+import static io.prestosql.jdbc.TestPrestoDriver.closeQuietly;
+import static java.lang.String.format;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertNotEquals;
+import static org.testng.Assert.assertTrue;
+
+/**
+ * Checks that JDBC dates and timestamps are interpreted in the client JVM timezone and should not
+ * be affected by the session timezone.  We also compare the behavior to a reference JDBC driver.
+ */
+@Test(singleThreaded = true)
+public class TestJdbcResultSetTimezone
+{
+    private static final String OTHER_TIMEZONE = "Asia/Kathmandu";
+
+    private TestingPrestoServer server;
+    private List<ReferenceDriver> referenceDrivers;
+
+    private Connection connection;
+    private Statement statement;
+
+    @BeforeClass
+    public void setupServer()
+    {
+        assertNotEquals(OTHER_TIMEZONE, TimeZone.getDefault().getID(), "We need a timezone different from the default JVM one");
+        Logging.initialize();
+        server = TestingPrestoServer.create();
+        referenceDrivers = ImmutableList.of(new PostgresqlReferenceDriver()/*, new OracleReferenceDriver()*/);
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void tearDownServer()
+    {
+        closeQuietly(server);
+        referenceDrivers.forEach(ReferenceDriver::close);
+    }
+
+    @SuppressWarnings("JDBCResourceOpenedButNotSafelyClosed")
+    @BeforeMethod
+    public void setup()
+            throws Exception
+    {
+        connection = DriverManager.getConnection("jdbc:presto://" + server.getAddress(), "test", null);
+        statement = connection.createStatement();
+        referenceDrivers.forEach(ReferenceDriver::setup);
+    }
+
+    @AfterMethod(alwaysRun = true)
+    public void teardown()
+            throws IOException
+    {
+        closeQuietly(statement);
+        closeQuietly(connection);
+        try (Closer closer = Closer.create()) {
+            for (ReferenceDriver driver : referenceDrivers) {
+                closer.register(driver::teardown);
+            }
+        }
+    }
+
+    @DataProvider
+    public Object[][] timeZoneIds()
+    {
+        return new Object[][] {
+                {"UTC"},
+                {"Europe/Warsaw"}
```

Comment:
Add also `ZoneId.systemDefault().getId()`

and some other zone west of Greenwich (even though ZoneId.systemDefault().getId() is west of Greenwich)